x/upgrade/keeper/keeper.go

```diff
@@ -84,7 +85,16 @@ func (k Keeper) SetModuleVersionMap(ctx sdk.Context, vm module.VersionMap) {
 	if len(vm) > 0 {
 		store := ctx.KVStore(k.storeKey)
 		versionStore := prefix.NewStore(store, []byte{types.VersionMapByte})
-		for modName, ver := range vm {
+		// for deterministic iteration order
+		sortedModNames := make([]string, 0, len(vm))
+
+		for key := range vm {
+			sortedModNames = append(sortedModNames, key)
+		}
+		sort.Strings(sortedModNames)
```

Comment:
Copying here my Discord msg: IMO the store here is always a cachekv store, which [sorts keys](https://github.com/cosmos/cosmos-sdk/blob/1178626e50b6c6a60149aa6cd14017607f98cd99/store/cachekv/store.go#L116), so write ordering doesn't matter. I also created a [small gist](https://gist.github.com/amaurym/5d3caa88b8ac4bd882eb994d92e723bc) to test this.

So IMO we don't need to sort here.

Reply:
I'd honestly just keep it there given that we got bitten. When we switch to using smt, this assumption will be hard to find hence return the non-determinism. I'd recommend we keep the sort.Strings.

Reply:
Sounds good. Let's add a state-breaking changelog entry.

Reply:
added a changelog entry